quickwit/quickwit-storage/src/local_file_storage.rs

```diff
@@ -177,17 +172,64 @@ impl Storage for LocalFileStorage {
 
     async fn delete(&self, path: &Path) -> StorageResult<()> {
         let full_path = self.root.join(path);
-        missing_file_is_ok(fs::remove_file(full_path).await)?;
-        let parent = path.parent();
-        if parent.is_none() {
-            return Ok(());
-        }
-        if let Err(error) = delete_all_dirs(self.root.to_path_buf(), parent.unwrap()).await {
-            warn!(error = ?error, path = %path.display(), "Failed to clean up parent directories.");
+        ignore_error_kind!(ErrorKind::NotFound, fs::remove_file(full_path).await)?;
+        if let Some(parent) = path.parent() {
+            if let Err(error) = delete_all_dirs(&self.root, parent).await {
+                warn!(error=?error, path=%path.display(), "Failed to delete directory.");
+            }
         }
         Ok(())
     }
 
+    async fn bulk_delete<'a>(&self, paths: &[&'a Path]) -> Result<(), BulkDeleteError> {
+        let mut successes = Vec::with_capacity(paths.len());
+        let mut failures = HashMap::new();
+        let mut parent_paths = HashSet::new();
+
+        let futures = paths
+            .iter()
+            .map(|path| async move {
+                let full_path = self.root.join(path);
+                let delete_res =
+                    ignore_error_kind!(ErrorKind::NotFound, fs::remove_file(full_path).await);
+                (path, delete_res)
+            })
+            .collect::<Vec<_>>();
+        let mut stream = futures::stream::iter(futures).buffer_unordered(10);
+
+        while let Some((path, delete_res)) = stream.next().await {
+            match delete_res {
+                Ok(_) => {
+                    successes.push(path.to_path_buf());
+
+                    if let Some(parent) = path.parent() {
+                        parent_paths.insert(parent);
+                    }
+                }
+                Err(error) => {
+                    let failure = DeleteFailure {
+                        error: Some(error.into()),
+                        ..Default::default()
+                    };
+                    failures.insert(path.to_path_buf(), failure);
+                }
+            }
+        }
+        for parent_path in parent_paths {
```

Comment:
Nitpick: I was suprised by the parent path stuff. I would add a doc on the function.

Reply:
Yep, I'm basically replicating what is currently done in `delete`, i.e. deleting empty parent directories and ignoring missing files.

Will add more doc.

Reply:
I added a proposition to relieve from that :)